core/font.lua

```diff
@@ -233,10 +227,10 @@ local font = {
       local font = ot.parseFont(face)
       if font.cpal then
          SILE.require("packages.color-fonts")
-         if SILE.shaper._name ~= "harfbuzzWithColor" then
+         if SILE.shaper._name ~= "harfbuzz-color" then
```

Comment:
We can rename it, but it won't make it work! (#2232)

Reply:
Besides, it's still and hard-coded reference to a shaper by name. Weird design.

Reply:
I agree it is a weird design, but there are several places where the shaper(s) reach into things they probably don't need. I was mainly focused on refactoring the language modules, not *all* the modules. Since we have a tracking issue for the broken color shaper and this isn't making the design *worse*, only focusing on isolating the language modules I'll leave this alone for now.